<!DOCTYPE html>
<title>Canvas test: 2d.drawImage.zerocanvas</title>
<script src="../tests.js"></script>
<link rel="stylesheet" href="../tests.css">
<link rel="prev" href="2d.drawImage.broken.html" title="2d.drawImage.broken">
<link rel="next" href="2d.drawImage.animated.gif.html" title="2d.drawImage.animated.gif">
<body class="show_output">
<p>
 <a href="2d.drawImage.broken.html" accesskey="p" title="[p] 2d.drawImage.broken">&lt;</a>
 <a href="index.html">[index]</a>
 <a href="2d.drawImage.animated.gif.html" accesskey="n" title="[n] 2d.drawImage.animated.gif">&gt;</a>
<h1><a href="index.2d.html">2d</a>.<a href="index.2d.drawImage.html">drawImage</a>.zerocanvas</h1>
<p class="desc"></p>
<div class="refs">References:
<ul>
<li><a href="spec.html#testrefs.2d.drawImage.zerocanvas">2d.drawImage.zerocanvas</a>

</ul>
</div>

<p class="output">Actual output:</p>
<canvas id="c" class="output" width="100" height="50"><p class="fallback">FAIL (fallback content)</p></canvas>
<p class="output expectedtext">Expected output:<p><img src="green-100x50.png" class="output expected" id="expected" alt="">
<ul id="d"></ul>
<script>
_addTest(function(canvas, ctx) {

ctx.fillStyle = '#0f0';
ctx.fillRect(0, 0, 100, 50);

var canvas2 = document.createElement('canvas');
canvas2.width = 0;
canvas2.height = 10;
try { var _thrown = false;
  ctx.drawImage(canvas2, 0, 0);
} catch (e) { if (e.code != DOMException.INVALID_STATE_ERR) _fail("Failed assertion: expected exception of type INVALID_STATE_ERR, got: "+e.message); _thrown = true; } finally { _assert(_thrown, "should throw exception of type INVALID_STATE_ERR: ctx.drawImage(canvas2, 0, 0)"); }

canvas2.width = 10;
canvas2.height = 0;
try { var _thrown = false;
  ctx.drawImage(canvas2, 0, 0);
} catch (e) { if (e.code != DOMException.INVALID_STATE_ERR) _fail("Failed assertion: expected exception of type INVALID_STATE_ERR, got: "+e.message); _thrown = true; } finally { _assert(_thrown, "should throw exception of type INVALID_STATE_ERR: ctx.drawImage(canvas2, 0, 0)"); }

canvas2.width = 0;
canvas2.height = 0;
try { var _thrown = false;
  ctx.drawImage(canvas2, 0, 0);
} catch (e) { if (e.code != DOMException.INVALID_STATE_ERR) _fail("Failed assertion: expected exception of type INVALID_STATE_ERR, got: "+e.message); _thrown = true; } finally { _assert(_thrown, "should throw exception of type INVALID_STATE_ERR: ctx.drawImage(canvas2, 0, 0)"); }

_assertPixelApprox(canvas, 50,25, 0,255,0,255, "50,25", "0,255,0,255", 2);


});
</script>

